src/saturn/sattypes.h

```diff
@@ -10,17 +10,42 @@ typedef unsigned short u16;
 typedef unsigned int u32;
 typedef unsigned long long u64;
 
-// entity layout is different somehow
 typedef struct Entity {
-    s16 temp;
-    s16 pad1;
+    void* unk0;
     s16 posX;
-    s16 pad3;
+    s16 pad0;
     s16 posY;
-    s8 pad[0x25];
+    s16 pad1[2];
+    s16 unke;
+    s16 pad2[2];
+    u16 unk10; // Hitbox X Offset
+    s16 unk12; // Hitbox Y Offset
+    s8 pad3[0x14];
+    u16 step;
+    u16 step_s;
     u16 subId;
+    u8 pad4[8];
+    u16 unk3C; // hitbox state
+    s16 pad5[4];
+    u8 hitboxWidth;
+    u8 hitboxHeight;
+    u8 unk48; // 1 = took hit
+    u8 pad6[0x4];
+    u16 animFrameIdx;
+    s16 animFrameDuration;
+    u8 pad7[0x24];
+    u16 unk74;
+    u8 pad8[0x24];
+    u16 zPriority;
+    u8 pad9[0x14];
+    u16 objectId;
+    u32 pad10;
 } Entity;
```

Comment:
```suggestion
typedef struct Entity {
    /* 0x00 */ void* unk0;
    /* 0x04 */ s16 posX;
    /* 0x06 */ char pad_6[0x2];
    /* 0x08 */ s16 posY;
    /* 0x0A */ char pad_A[0x4];
    /* 0x0E */ s16 unkE;
    /* 0x10 */ char pad_10[0x4];
    /* 0x14 */ u16 unk14; // Hitbox X Offset
    /* 0x16 */ s16 unk16; // Hitbox Y Offset
    /* 0x18 */ char pad_18[0x14];
    /* 0x2C */ u16 step;
    /* 0x2E */ u16 step_s;
    /* 0x30 */ u16 subId;
    /* 0x32 */ char pad_32[0x8];
    /* 0x3A */ u16 unk3A; // hitbox state
    /* 0x3C */ char pad_3C[0x8];
    /* 0x44 */ u8 hitboxWidth;
    /* 0x45 */ u8 hitboxHeight;
    /* 0x46 */ u8 unk46; // 1 = took hit
    /* 0x47 */ char pad_47[0x4];
    /* 0x4B */ u16 animFrameIdx;
    /* 0x4D */ s16 animFrameDuration;
    /* 0x4F */ char pad_4F[0x24];
    /* 0x73 */ u16 unk73;
    /* 0x75 */ char pad_75[0x24];
    /* 0x99 */ u16 zPriority;
    /* 0x9B */ char pad_9B[0x14];
    /* 0xAF */ u16 objectId;
    /* 0xB1 */ char pad_B1[0x4];
} Entity;

Reply:
`/* 0x73 */ u16 unk73; isn't correct`, that is at offset 0x74. There may be other differences. I was using the PSX names so that they can be correlated like with the hitbox offsets. I think these should be renamed in the PSX version since we know what they do.

Reply:
```suggestion
typedef struct Entity {
    /* 0x00 */ void* unk0;
    /* 0x04 */ s16 posX;
    /* 0x06 */ char pad_6[0x2];
    /* 0x08 */ s16 posY;
    /* 0x0A */ char pad_A[0x4];
    /* 0x0E */ s16 unkE;
    /* 0x10 */ char pad_10[0x4];
    /* 0x14 */ u16 unk14; // Hitbox X Offset
    /* 0x16 */ s16 unk16; // Hitbox Y Offset
    /* 0x18 */ char pad_18[0x14];
    /* 0x2C */ u16 step;
    /* 0x2E */ u16 step_s;
    /* 0x30 */ u16 subId;
    /* 0x32 */ char pad_32[0x8];
    /* 0x3A */ u16 unk3A; // hitbox state
    /* 0x3C */ char pad_3C[0x8];
    /* 0x44 */ u8 hitboxWidth;
    /* 0x45 */ u8 hitboxHeight;
    /* 0x46 */ u8 unk46; // 1 = took hit
    /* 0x47 */ char pad_47[0x5];
    /* 0x4C */ u16 animFrameIdx;
    /* 0x4E */ s16 animFrameDuration;
    /* 0x50 */ char pad_4F[0x24];
    /* 0x74 */ u16 unk74;
    /* 0x76 */ char pad_75[0x24];
    /* 0x9A */ u16 zPriority;
    /* 0x9C */ char pad_9B[0x14];
    /* 0xB1 */ u16 objectId;
    /* 0xB2 */ char pad_B1[0x4];
} Entity;

Reply:
Revised Entity struct with visible struct padding.